lib/msf_autoload.rb

```diff
@@ -333,3 +333,6 @@ def finalize_loader(loader)
 autoload :RubySMB, 'ruby_smb'
 
 require 'rexml/document'
+
+# XXX: Should be removed once the `lib/metasploit` folder is loaded by Zeitwerk
+require 'metasploit/framework/hashes'
```

Comment:
This will probably look something like:

```diff
diff --git a/lib/msf_autoload.rb b/lib/msf_autoload.rb
index 7bb1da35af..0b697e6ae0 100644
--- a/lib/msf_autoload.rb
+++ b/lib/msf_autoload.rb
@@ -298,6 +298,7 @@ class MsfAutoload
     [
       { path: "#{__dir__}/msf/", namespace: Msf },
       { path: "#{__dir__}/rex/", namespace: Rex },
+      { path: "#{__dir__}/metasploit/", namespace: Metasploit }
     ]
   end
```

But it would require more testing/fixing than the cycles currently available